kernel/frame_allocator/src/lib.rs

```
@@ -382,24 +572,61 @@ impl AllocatedFrames {
             return Err(other);
         }
 
-        // ensure the now-merged AllocatedFrames doesn't run its drop handler and free its frames.
+        // ensure the now-merged Frames doesn't run its drop handler
         core::mem::forget(other); 
         Ok(())
     }
 
-    /// Splits this `AllocatedFrames` into two separate `AllocatedFrames` objects:
+    /// Splits up the given `Frames` into multiple smaller `Frames`.
+    /// 
+    /// Returns a tuple of three `Frames`:
+    /// 1. The `Frames` containing the requested range of frames starting at `start_frame`.
+    /// 2. The range of frames in the `self` that came before the beginning of the requested frame range.
+    /// 3. The range of frames in the `self` that came after the end of the requested frame range.
+    /// 
+    /// If `start_frame` is not contained within `self` or `num_frames` results in an end frame greater than the end of `self`,
+    /// then `self` is not changed and we return (self, None, None).
```

this seems like a code smell (design mistake) in that you're hiding errors. This condition should be returned as an explicit error, with a function signature like
```rust
pub fn split_range(self, range: FrameRange) -> Result<SplitRange, Self> { ... }
```